lightning/src/ln/channel.rs

```diff
@@ -1719,9 +1795,50 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 				removed_outbound_total_msat += htlc.amount_msat;
 			}
 		}
-		if htlc_inbound_value_msat + msg.amount_msat + self.value_to_self_msat > (self.channel_value_satoshis - Channel::<ChanSigner>::get_remote_channel_reserve_satoshis(self.channel_value_satoshis)) * 1000 + removed_outbound_total_msat {
-			return Err(ChannelError::Close("Remote HTLC add would put them under their reserve value"));
+
+		if self.channel_outbound {
+			// Check that they won't violate our local required channel reserve by adding this HTLC.
+
+			// +1 for this HTLC.
+			let local_fee_cost_msat = self.next_local_commit_tx_fee_msat(Some(1));
+			if self.value_to_self_msat < self.local_channel_reserve_satoshis * 1000 + local_fee_cost_msat {
```

Comment:
I think not because the HTLC that they're paying us hasn't gone through the whole removal process yet, so it can't count towards our balance w/r/t reserve calculations.

Reply:
Hm, I'm not sure on this one actually... because we do count pending HTLCs towards our/the remote balance in other places. Looking into it

Reply:
No, I'm pretty sure it's right. It's basically checking, if this HTLC _were_ to be included in our commitment tx, and that commitment tx were broadcasted before the HLTC was removed (aka before the HTLC's value could be counted towards our onchain fees), would we be able to cover the commit tx fee.

It's pretty complicated so LMK if that doesn't sound right...